src/instructlab/data/generate.py

```diff
@@ -214,10 +222,25 @@ def generate(
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
         # TODO (cdoern): we really should not edit the cfg object
-        ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(
-            cfg=ctx.obj.config.generate.teacher, model_path=model_path
+        gen_cfg = copy.copy(ctx.obj.config)
+        gen_cfg.generate.teacher.llama_cpp.llm_family = model_family
+        if gpus is not None:
+            tps_prefix = "--tensor-parallel-size"
+            if backends.vllm.contains_argument(
+                tps_prefix, gen_cfg.generte.teacher.vllm.vllm_args
+            ):
+                click.secho(
+                    "Using gpus from --gpus or generate.teacher config and ignoring --tensor-parallel-size configured in generate.teacher vllm_args",
+                    fg="yellow",
+                )
+            gen_cfg.generate.teacher.vllm.vllm_args.extend([tps_prefix, str(gpus)])
+        backend_instance, backend_type = backends.select_backend(
```

Comment:
We should just use `is_model_safe_tensors()` here on the model that's being served if gpus is set. If that returns false then throw that log message. No need to change `select_backend`.

 https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/backends/backends.py#L115

Reply:
I disagree, I'd like to get a 100% y/n on this rather than just saying "oh its safetensors".

I also need to do this any way for my other GPU related PR in ilab model serve.

This seems like a nit as well IMO

Reply:
Would it be better to have the backend have a method to return its type?

Reply:
After looking deeper into `select_backend()`, it's calling `get()` and that is using `is_model_safe_tensors()` under the hood along with more robust model type checking. 

It seems like that would be the method we'd want to use to determine it's type, but `select_backend()` sets the path argument for the call to `get()` so I don't think we can use it as a drop in replacement to what Charlie did yet.

Reply:
> Would it be better to have the backend have a method to return its type?

To add a little more detail.  Rather than return an object and a string identifier for that object.  Why not have the instance be able to identify itself.  You could already use isinstance.  But if there was a backend_type() method (or something similar), you could avoid returning the two separate objects to represent one thing.

Reply:
sure I can opt for that approach @danmcp